### ISSUE TITLE
Unable to disable default mappings for commit editor

### ISSUE THREAD

**Issue by trimclain** [NeogitOrg/neogit#1130]
### Description

I couldn't  disable the default mappings in `commit_editor` the same way as in `popup` section. For some reason it breaks the commit editor popup without any error messages.

### Neovim version

NVIM v0.10.0-dev-1223+gde9348978
Build type: Release
LuaJIT 2.1.1695653777

### Operating system and version

Ubuntu 22.04.3

### Steps to reproduce

1. `nvim -nu minimal.lua`
2. `:Neogit`
3. Have at least one staged change
4. Use `c` and then `c` to open the commit editor popup

### Expected behavior

It opens the commit editor popup. The specified keys are disabled.

### Actual behavior

The commit editor doesn't open (or opens and closes fast, I'm not sure) without any error messages.

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({
    -- disable_insert_on_commit = true, -- "auto", "true" or "false"
    mappings = {
        commit_editor = {
            ["<c-c><c-c>"] = false, -- default: "Submit"
            ["<c-c><c-k>"] = false, -- default: "Abort"
        },
    },
}) -- For instance, setup Neogit
```

**Comment by davidsierradz** [comment]
Is this fixed? I'm in nvim 0.10.0 and neogit@5b7bad8 (latest of master) and when trying to disable some mappings:

```
require('neogit').setup({
  mappings = {
    commit_editor = {
      ['q'] = false,
    },
    rebase_editor = {
      ['q'] = false,
    },
    status = {
      ['q'] = false,
    },
  },
})
```

When opening Neogit with `:Neogit`, get the following:

```
Error executing Lua callback: ...hare/nvim/lazy/neogit/lua/neogit/buffers/status/init.lua:130: table index is nil
stack traceback:
	...hare/nvim/lazy/neogit/lua/neogit/buffers/status/init.lua:130: in function 'open'
	.../neuromante/.local/share/nvim/lazy/neogit/lua/neogit.lua:120: in function 'open_status_buffer'
	.../neuromante/.local/share/nvim/lazy/neogit/lua/neogit.lua:192: in function 'open'
	...uromante/.local/share/nvim/lazy/neogit/plugin/neogit.lua:5: in function <...uromante/.local/share/nvim/lazy/neogit/plugin/neogit.lua:3>
```

Thanks!
